### PR TITLE
Conformer End-to-End Speech Translation

### PR DESCRIPTION
**Pull request by hirofumi0810**

### CODE REVIEW

**Comment by sw005320** [comment]
Many thanks!
@pengchengguo, could you review this PR?

**Review by sw005320 — APPROVED** [review]
LGTM
- Once you get a result, please update the result and upload a model.
- You can make these configurations as default in the fisher_callhome_spanish task (change the symlink target to conformer one).

File: espnet/nets/pytorch_backend/e2e_st_conformer.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
```
Comment (ShigekiKarita) [inline espnet/nets/pytorch_backend/e2e_st_conformer.py:1]: shebang is not needed for non executables

File: espnet/nets/pytorch_backend/e2e_st_conformer.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        """Construct an E2E object.
+
+        :param int idim: dimension of inputs
+        :param int odim: dimension of outputs
+        :param Namespace args: argument Namespace containing options
+        """
+        super().__init__(idim, odim, args, ignore_id)
+        if args.transformer_attn_dropout_rate is None:
+            args.transformer_attn_dropout_rate = args.dropout_rate
+        self.encoder = Encoder(
```
Comment (ShigekiKarita) [inline espnet/nets/pytorch_backend/e2e_st_conformer.py:55]: How about make this encoder initialization in one common function like the `add_conformer_arguments`?

Comment (hirofumi0810) [reply espnet/nets/pytorch_backend/e2e_st_conformer.py:52]: I think that is a general refactoring for all encoder-decoder models, so I'd like to consider it in another PR.

**Comment by hirofumi0810** [comment]
@sw005320 I change the config link to the Conformer one.
Let me upload models in another PR because the Conformer models are still under tuning and maybe I will frequently change the config until then.

**Comment by sw005320** [comment]
Many thanks for the impressive results!!!

> Let me upload models in another PR because the Conformer models are still under tuning and maybe I will frequently change the config until finishing tuning.

Sure, this sounds good to me!

**Pull request merged**